components/builder-api-client/src/lib.rs

```diff
@@ -197,6 +199,86 @@ pub struct OriginChannelIdent {
     pub name: String,
 }
 
+#[derive(Clone, Deserialize)]
+pub struct OriginInvitation {
+    #[serde(with = "json_u64")]
+    pub id: u64,
+    #[serde(with = "json_u64")]
+    pub account_id: u64,
+    pub account_name: String,
+    pub created_at: String,
+    pub ignored: bool,
+    pub origin: String,
+    #[serde(with = "json_u64")]
+    pub owner_id: u64,
+    pub updated_at: String,
+}
+
+#[derive(Clone, Deserialize)]
+pub struct UserOriginInvitationsResponse(pub Vec<OriginInvitation>);
+
+#[derive(Clone, Deserialize)]
+pub struct PendingOriginInvitationsResponse {
+    pub origin:      String,
+    pub invitations: Vec<OriginInvitation>,
+}
+
+macro_rules! invitation_format {
```

Comment:
This is pretty clever. I had to sit here for a moment and think about whether or not i liked the decision to use a macro here but ultimately its pretty clean and I'm fairly ambivalent on whether its the right or wrong thing. I'm curious how you might solve the problem without a macro but no obligation to do so from me.

Reply:
Yes, the macro might have been clever but, alas, merited a more robust implementation. I've removed the macro and introduced a Tabular trait in https://github.com/habitat-sh/habitat/pull/7349/commits/a65aebfafec7355e188e527768485ba9957fa404